libs/icu_headroom_metric.py

```diff
@@ -100,7 +100,7 @@ def _latest_icu_beds(self):
     @property
     def total_icu_beds(self) -> Union[pd.Series, float]:
         timeseries = self._data[CommonFields.ICU_BEDS]
-        latest_icu_beds = self._latest_icu_beds
+        latest_icu_beds = self.latest_icu_beds
 
         if timeseries.any():
             return timeseries.fillna(latest_icu_beds)
```

Comment:
FWIW- this latest_icu_beds / total_icu_beds code is kinda' confusing:

1. Why does this return a Series / float union?  Wouldn't it be cleaner to make it always return a series?
2. Why do we fill gaps with the latest value?  That seems kinda' non-sensical and could result in weird jumps, right?  If we really want to try to keep whatever gappy data we have, then we should probably interpolate over the gaps or something?
3. In cases where we have two conceptually similar methods that just differ in their return type (latest_icu_beds and total_icu_beds), it would be nice to name them to match, e.g. total_icu_beds_latest and total_icu_beds_series or something?  Idunno.